main.tf

```diff
@@ -0,0 +1,84 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 2.92"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_group" "main" {
+  name = "Softwire21_AliceWenban_ProjectExercise"
+}
+
+resource "azurerm_app_service_plan" "main" {
+  name                = "${var.prefix}-terraformed-asp"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  kind                = "Linux"
+  reserved            = true
+
+  sku {
+    tier = "Basic"
+    size = "B1"
+  }
+}
+
+resource "azurerm_cosmosdb_account" "main" {
+  name                = "${var.prefix}-cosmos-db-account"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  offer_type          = "Standard"
+  kind                = "MongoDB"
+
+  capabilities {
+    name = "EnableMongo"
+  }
+
+  capabilities {
+    name = "EnableServerless"
+  }
+
+  consistency_policy {
+    consistency_level = "BoundedStaleness"
+  }
+
+  geo_location {
+    location          = var.location
+    failover_priority = 0
+  }
+
+  lifecycle { prevent_destroy = true }
+}
+
+resource "azurerm_cosmosdb_mongo_database" "main" {
+  name                = "${var.prefix}-cosmos-mongo-db"
+  resource_group_name = azurerm_cosmosdb_account.main.resource_group_name
+  account_name        = azurerm_cosmosdb_account.main.name
+}
+
+
+resource "azurerm_app_service" "main" {
+  name                = "${var.prefix}-todo-app"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  app_service_plan_id = azurerm_app_service_plan.main.id
+
+  site_config {
+    linux_fx_version = "DOCKER|aliwen/todo_app:latest"
+  }
+
+  app_settings = {
+    "DOCKER_REGISTRY_SERVER_URL" = "https://aliwen"
```

Comment:
This doesn't feel right; are you actually running a registry at that server?

Separately, one for our instructions to update, but worth noting anyway that they recently deprecated this so the newly recommended approach is [using `azurerm_linux_web_app` instead](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_web_app)